From 23da6073369ef6a99d1789de967e57359d1fe3b0 Mon Sep 17 00:00:00 2001
From: Stefan Dirsch <sndirsch@suse.de>
Date: Tue, 26 Apr 2016 16:26:49 +0200
Subject: Disable "acceleration" under qemu

Disable "acceleration" under qemu, since taking the hypercall trap
is really quite expensive and you're better off doing noaccel. Patch
is based on the one by "Adam Jackson" <ajax@redhat.com>

http://pkgs.fedoraproject.org/cgit/rpms/xorg-x11-drv-cirrus.git/tree/cirrus-1.2.0-qemu.patch?id=daccd1c8174623500eddfa297d8ea76a86d3c5d9

Reviewed-by: Adam Jackson <ajax@redhat.com>

diff --git a/src/alp_driver.c b/src/alp_driver.c
index bd5e52f..ba55247 100644
--- src/alp_driver.c
+++ src/alp_driver.c
@@ -774,6 +774,16 @@ AlpPreInit(ScrnInfoPtr pScrn, int flags)
      else
  	xf86SetDDCproperties(pScrn,xf86PrintEDID(
 		 xf86DoEDID_DDC2(XF86_SCRN_ARG(pScrn),pCir->I2CPtr1)));
+
+#ifdef XSERVER_LIBPCIACCESS
+     #ifndef PCI_CHIP_QEMU
+     #define PCI_CHIP_QEMU 0x1af4
+     #endif
+     if (!pScrn->monitor->DDC &&
+	((pCir->PciInfo->subvendor_id & 0xffff) == PCI_CHIP_QEMU)) {
+	pCir->NoAccel = TRUE;
+     }
+#endif
  
      /* Probe the possible LCD display */
      AlpProbeLCD(pScrn);
-- 
cgit v0.10.2

